nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/text/TextLineReader.java

```diff
@@ -0,0 +1,127 @@
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .defaultValue("\\n")
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor LINES_PER_RECORD = new PropertyDescriptor.Builder()
+            .name("linereader-lines-per-record")
+            .displayName("Lines Per Record")
+            .description("The number of lines that will be added to the field's value. This allows groups of lines to be treated as a single string for the purposes of populating the field value. "
+                    + "To split each line into its own record, set this value to 1.")
+            .required(true)
+            .defaultValue("1")
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.POSITIVE_INTEGER_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor SKIP_LINE_COUNT = new PropertyDescriptor.Builder()
+            .name("linereader-skip-count")
+            .displayName("Skip Line Count")
+            .description("The number of lines at the beginning of the input that should be ignored and thus not treated as records.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .defaultValue("0")
+            .build();
+
+    protected static final PropertyDescriptor LINE_FIELD_NAME = new PropertyDescriptor.Builder()
+            .name("linereader-field-name")
+            .displayName("Output Field Name")
+            .description("Specifies the name of the record field in the output, whose values will be the text lines from the input as configured in the processor.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    protected static final PropertyDescriptor IGNORE_EMPTY_LINES = new PropertyDescriptor.Builder()
+            .name("linereader-ignore-empty-lines")
+            .displayName("Ignore Empty Lines")
+            .description("If this property is true, then any empty lines in the input will not be treated as part of the record field value. If this property is false, "
+                    + "an empty line is considered a line. If Lines Per Record is set to 1 and this property is false, the record field will contain the empty string.")
+            .expressionLanguageSupported(ExpressionLanguageScope.NONE)
+            .allowableValues("true", "false")
+            .defaultValue("false")
+            .required(true)
+            .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
```

Comment:
It would be a bit more efficient to create this all at once with a static constructor such as

`public static final List<PropertyDescriptor> PROPS = Collections.unmodifiableList(Arrays.asList(.....));`

Reply:
@mattyb149 can you address this? or reply?

Reply:
This is done differently by different processors, I copied this part of the code from some other RecordReader.